pkg/controller/platformadmin/platformadmin_controller.go

```diff
@@ -600,3 +611,111 @@ func annotationToComponent(annotation map[string]string) ([]*config.Component, e
 
 	return components, nil
 }
+
+func NewYurtIoTCarrierComponent(platformAdmin *iotv1alpha2.PlatformAdmin) (*config.Component, error) {
+	var yurtIotCarrierComponent config.Component
+	var yurtIoTCarrierDeployment iotv1alpha1.DeploymentTemplateSpec
+	ver, ns, err := util.DefaultVersion(platformAdmin)
+	if err != nil {
+		return nil, err
+	}
+	// YurtIoTCarrier doesn't need a service yet
+	yurtIoTCarrierConfig := map[string]interface{}{
```

Comment:
Here don't use the `map[string]interface`, please use `appsv1.DeploymentSpec` and `appsv1alpha1.YurtAppSet`.